File: api/src/models/job.ts
/**
 * id
 * name
 * description
 * date_published
 * job_category_Id
 * company_id
 *
 */

import mongoose, { Document, Schema } from "mongoose";

export interface IJob extends Document {
  name: String;
  description: String;
  job_position: String;
  experience_level: String;
  other_info: String;
  deadline: String;
  date_published: String;
  job_category_id: mongoose.Types.ObjectId;
  job_type: JobTypes;
  company_id: String;
}

export enum JobTypes {
  INTERNSHIP,
  CONTRACT,
}

const jobSchema: Schema<IJob> = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
    },
    description: {
      type: String,
      required: true,
    },
    job_position: { type: String, default: "" },
    experience_level: { type: String, required: true },
    other_info: { type: String, default: "" },
    deadline: { type: Date, required: true },
    date_published: {
      type: Date,
      default: Date.now(),
    },
    job_type: {
      type: String,
      default: "CONTRACT",
    },
    job_category_id: {
      type: mongoose.Types.ObjectId,
      ref: "JobCategory",
    },
    company_id: {
      type: mongoose.Types.ObjectId,
      ref: "User",
    },
  },
  { timestamps: true }
);

export default mongoose.model("Job", jobSchema);

File: app/lib/models/job.dart
import "package:meta/meta.dart";
import 'package:equatable/equatable.dart';

class Job extends Equatable {
  final String id;
  final String name;
  final String description;
  final String jobPosition;
  final String otherInfo;
  final String experienceLevel;
  final DateTime deadline;
  final String categoryId;
  final String jobType;
  final DateTime datePublished;
  final String companyId;

  Job({
    this.id,
    @required this.name,
    @required this.description,
    @required this.jobPosition,
    @required this.otherInfo,
    @required this.experienceLevel,
    @required this.deadline,
    @required this.categoryId,
    @required this.jobType,
    @required this.datePublished,
    @required this.companyId,

  });


  factory Job.fromJson(Map<String, dynamic> json) {
    return Job(
      id: json['_id'],
      name: json['name'],
      description: json['description'],
      jobPosition: json['job_position'],
      otherInfo: json['other_info'],
      experienceLevel: json['experience_level'],
      deadline: DateTime.parse(json['deadline']),
      categoryId: json['job_category_id'],
      jobType: json['job_type'],
      datePublished: DateTime.parse(json['date_published']),
      companyId: json['company_id'],
    );
  }

  @override
  List<Object> get props => [
        id,
        name,
        description,
        jobPosition,
        otherInfo,
        experienceLevel,
        deadline,
        jobType,
        categoryId,
        datePublished,
        companyId
      ];
}
